rstream/producer.py

```diff
@@ -146,9 +163,23 @@ async def _get_or_create_publisher(
             partial(self._on_publish_error, publisher=publisher),
             name=publisher.reference,
         )
+
+        if send_batch_enabled is True and self.thread_active is False:
```

Comment:
For the moment I moved it in the Producer constructor thinking it could help also with the other open issue (but apparently isn't). The disadvantage is that the thread will be run also if the client will not use they asynchronous API. Alternatively can be put inside the send() and activated just the first time it is invoked.

Reply:
I think we should start this task either inside `start()` method or lazily inside `send()`, as you said.

Reply:
I put it lazily in the send() then. i think it's better to keep things coherent with the other clients too!